horizomer/utils/tree.py

```diff
@@ -44,6 +44,31 @@ def compare_topology(tree1, tree2):
     return n2p1 == n2p2
 
 
+def intersect_trees(tree1, tree2):
+    """Shrink two trees to contain only overlapping taxa.
+
+    Parameters
+    ----------
+    tree1 : skbio.TreeNode
+        first tree to intersect
+    tree2 : skbio.TreeNode
+        second tree to intersect
+
+    Returns
+    -------
+    tuple of two TreeNodes
+        resulting trees containing only overlapping taxa
+    """
+    taxa1 = set([tip.name for tip in tree1.tips()])
+    taxa2 = set([tip.name for tip in tree2.tips()])
+    taxa_lap = taxa1.intersection(taxa2)
+    if len(taxa_lap) == 0:
+        raise ValueError('Trees have no overlapping taxa.')
+    tree1_lap = tree1.shear(taxa_lap)
```

Comment:
shouldn't you also call http://scikit-bio.org/docs/0.5.1/generated/skbio.tree.TreeNode.prune.html#skbio.tree.TreeNode.prune after shearing?

Reply:
scikit-bio's [shear function](https://github.com/biocore/scikit-bio/blob/master/skbio/tree/_tree.py#L470) already does prune in the end.

Reply:
ups, than I can save some time in some of my functions :-)